elife/nginx.sls

```diff
@@ -94,6 +94,11 @@ nginx-server-service:
             # this sounds like a good thing, but some conf files reference external files that may not be 
             # ready when they appear in here. explicit requisites/watches are better.
             #- file: /etc/nginx/sites-enabled/*
+{% if salt['elife.cfg']('cfn.outputs.DomainName') %}
```

Comment:
I discovered this file includes `elife.certificates`: https://github.com/elifesciences/builder-base-formula/blob/5d8102bed07710a7bc3a41d8c57576c789c0c0c7/elife/nginx.sls#L75
If this is correct, then `elife.certificates` should be removed from https://github.com/elifesciences/builder-private/blob/master/salt/top.sls#L33 right?

The `if` becomes the same as https://github.com/elifesciences/builder-base-formula/pull/203/files#diff-35caf6f7a7a31050fce46cdeb1e29a5fL24 so it should be safe for now. However could we do this with a `listen_in` from `web-complete-cert`? This is how [this file does it](https://github.com/elifesciences/builder-base-formula/blob/5d8102bed07710a7bc3a41d8c57576c789c0c0c7/elife/nginx.sls#L23) and it could work without this `if`.

Reply:
that's possible, but certificates.sls has no dependency on any webserver or it's configuration. This was back when we had Apache and Nginx in play. Now that civi and the last app using Apache has been removed the distinction doesn't seem that important any more. 

Can you think of a case where we would need the SSL certificates on a server but not a web server? An FTP or mail server or something?

The master-server project has a dependency on certificates.sls as well, not sure why.

Reply:
We do use them with ELB/CloudFront/Fastly but Salt isn't involved in that. 

The corner cases that come to mind are:
- local usage: web servers are there but no certificates (or fake ones maybe) are used
- [https://github.com/elifesciences/redirects-formula/blob/master/salt/example.top](redirects) which uses its own certificates because it has different domains to cover. This one pushes for decoupling of the two more than for the use of `elife.certificates` without `elife.nginx`.

Reply:
Another case:
- `elife.vault` relies on `elife.certificates` only, https://github.com/elifesciences/master-server-formula/blob/27cf38250da0a339cbb591d400b00249ca0bbdec/salt/master-server/config/etc-vault.hcl
so the direction of the dependency is correct `elife.nginx`, depends on `elife.certificates`

Reply:
I'll merge this, but we could consider self-signed certificates to get rid of that great big `if` statement and ensure a certificate is always there